ruby/private/bundle/bundle.bzl

```diff
@@ -1,123 +1,162 @@
+BUNDLE_BINARY = "bundler/exe/bundler"
+SCRIPT_INSTALL_GEM = "install_bundler.rb"
+SCRIPT_BUILD_FILE_GENERATOR = "create_bundle_build_file.rb"
 
-    result = ctx.execute(args, environment = environment)
-    if result.return_code:
-        message = "Failed to evaluate ruby snippet with {}: {}".format(
-            interpreter,
-            result.stderr,
-        )
-        fail(message)
+def run_bundler(runtime_ctx, extra_args):
+    # Now we are running bundle install
+    args = [
+        runtime_ctx.interpreter,  # ruby
+        "--enable=gems",  # bundler must run with rubygems enabled
+        "-I",
+        ".",
+        "-I",  # Used to tell Ruby where to load the library scripts
+        BUNDLE_INSTALL_PATH,  # Add vendor/bundle to the list of resolvers
+        BUNDLE_BINARY,  # our binary
+    ] + extra_args
 
-def bundle_install_impl(ctx):
-    ctx.symlink(ctx.attr.gemfile, "Gemfile")
-    ctx.symlink(ctx.attr.gemfile_lock, "Gemfile.lock")
-    ctx.symlink(ctx.attr._create_bundle_build_file, "create_bundle_build_file.rb")
-    ctx.symlink(ctx.attr._install_bundler, "install_bundler.rb")
+    # print("running bundler with args\n", args)
 
-    # TODO(kig) Make Gemspec reference from Gemfile actually work
-    if ctx.attr.gemspec:
-        ctx.symlink(ctx.attr.gemspec, ctx.path(ctx.attr.gemspec).basename)
+    result = runtime_ctx.ctx.execute(
+        args,
+        quiet = False,
+        environment = runtime_ctx.environment,
+    )
 
-    ruby = ctx.attr.ruby_interpreter
-    interpreter_path = ctx.path(ruby)
+    return result
 
-    install_bundler(
-        ctx,
-        interpreter_path,
-        "install_bundler.rb",
-        "bundler",
-        ctx.attr.version,
-    )
+# Sets local bundler config values
+def set_bundler_config(runtime_ctx):
+    # Bundler is deprecating various flags in favor of the configuration.
+    # HOWEVER — for reasons I can't explain, Bazel runs "bundle install" *prior*
+    # to setting these flags. So the flags are then useless until we can force the
+    # order and ensure that Bazel first downloads Bundler, then sets config, then
+    # runs bundle install. Until then, it's a wild west out here.
+    #
+    # Set local configuration options for bundler
+    bundler_config = {
+        "binstubs": "bin",
+        "deployment": "'true'",
+        "standalone": "'true'",
+        "frozen": "'true'",
+        "without": "development,test",
+        "path": "lib",
+        "jobs": "20",
+    }
 
-    bundler = Label("//:bundler/exe/bundler")
+    for option, value in [(option, value) for option, value in bundler_config.items()]:
+        args = ["config", "--local", value, option]
 
-    # Install the Gems into the workspace
+        result = run_bundler(runtime_ctx, args)
+        if result.return_code:
+            message = "Failed to set bundle config {} to {}: {}".format(
+                option,
+                value,
+                result.stderr,
+            )
+            fail(message)
+
```

Comment:
The reason it’s only installing pure ruby gems is only due to the script we use which does unpack only, not instal.